resources/store/index.js

```diff
@@ -55,36 +55,75 @@ function processInitialData( data ) {
 	} );
 }
 
+/**
+ * Helper function to ensure the user doesn't try to access an invalid tab.
+ *
+ * @param {Object} state
+ * @param {string} tab
+ */
+function ensureTabExists( state, tab ) {
+	var tabs = Object.keys( state.images );
+
+	if ( tabs.indexOf( tab ) === -1 ) {
+		throw new Error( 'invalid tab' );
```

Comment:
That's a good idea but it might make sense to do it at the level of the caller. This helper function is used in both actions and mutations, and the way you'd want to switch tabs might be different in each context.

Also, I was mainly concerned here with a sanity check, now that we've introduced the complexity of having multiple queues. I really don't want to get tripped up because of a difference in casing between `User` and `user` at some point, so having an explicit error for now seemed useful.

Reply:
Yeah, that makes sense. I'm a little concerned we're going to miss some functionality (e.g. a previous bug fix) as we're porting things over piecemeal, so maybe we just need to think about how we can QA this thing to make sure we've covered everything at the end?